lightning/src/ln/channelmanager.rs

```diff
@@ -7678,6 +7749,38 @@ mod tests {
 		// Check that using the original payment hash succeeds.
 		assert!(inbound_payment::verify(payment_hash, &payment_data, nodes[0].node.highest_seen_timestamp.load(Ordering::Acquire) as u64, &nodes[0].node.inbound_payment_key, &nodes[0].logger).is_ok());
 	}
+
+	#[test]
+	fn sent_probe_is_probe_of_sending_node() {
+		let chanmon_cfgs = create_chanmon_cfgs(3);
+		let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, None, None, None]);
+		let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+
+		create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known()).0.contents.short_channel_id;
+		create_announced_chan_between_nodes(&nodes, 1, 2, InitFeatures::known(), InitFeatures::known()).0.contents.short_channel_id;
+
+		let (route, _, _, _) = get_route_and_payment_hash!(&nodes[0], nodes[2], 100000);
+
+		match nodes[0].node.send_probe_payment(&route) {
+			Ok((payment_hash, payment_id)) => {
+				assert!(nodes[0].node.payment_is_probe(payment_hash, payment_id));
+				assert!(!nodes[1].node.payment_is_probe(payment_hash, payment_id));
+				assert!(!nodes[2].node.payment_is_probe(payment_hash, payment_id));
+			},
+			_ => panic!(),
+		}
```

Comment:
Could we test the full probe flow, checking that the correct events are generated?

Reply:
Added a full end-to-end test with https://github.com/lightningdevkit/rust-lightning/pull/1567/commits/f13eed9c6e5cdfe1495d8d63ff5a1721a2a9cf07.